pkg/asset/machines/machineconfig/gcp/redirectports.go

```diff
@@ -0,0 +1,99 @@
+package gcp
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"text/template"
+
+	"github.com/pkg/errors"
+
+	"github.com/coreos/ignition/config/util"
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	"github.com/openshift/installer/data"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	scriptTemplate = "bootstrap/gcp/files/usr/local/bin/redirect-ports.sh.template"
+	scriptFile     = "/usr/local/bin/redirect-ports.sh"
+	unitFile       = "bootstrap/gcp/systemd/units/redirect-ports.service"
+)
+
+// ForPortRedirection creates the MachineConfig that adds a port redirection systemd unit
+// to machines on GCP. For masters, it redirects incoming traffic. For all nodes it redirects
+// outgoing traffic to the internal control plane load balancer.
+func ForPortRedirection(role string, clusterDomain string) (*mcfgv1.MachineConfig, error) {
+	templateData := map[string]interface{}{
+		"ClusterDomain": clusterDomain,
+		"ControlPlane":  (role == "master"),
+	}
+	script, err := readTemplate(scriptTemplate, templateData)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot read %s", scriptFile)
+	}
+	unit, err := readFile(unitFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot read %s", unitFile)
+	}
+	return &mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("99-%s-redirect-lb-ports", role),
```

Comment:
Any MachineConfig objects created by the installer are "unmanaged"; we don't have an owner for them or a way to upgrade.

Related: https://github.com/openshift/machine-config-operator/pull/1031

I think this script should probably just be shipped in RHCOS by default, and the MCO templates configuration for it.  It's somewhat like what we're doing in the MCO for OpenStack and bare metal right?

Reply:
Thanks @cgwalters makes sense. At this point, I'm not sure that this is the direction we want to go in, and makes it easier to test if it's self-contained in the installer. I will move it there if we agree that this is what we want to do.

Reply:
As far as testing, I totally get it!  It's not too bad to build your own release image though.